Updated Dapr Agents docs to align with recent quickstart changes

bibryam opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within tabpane
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have tabpane

In addition, please fill out the following to help reviewers understand this pull request:

## Description

* **Getting Started guide**: Reworked to align with the latest Dapr Agents quickstarts, using `06_durable_agent_http.py` as the primary walkthrough. Updated the flow to cover cloning the repo, environment setup, durable agent creation, Dapr component configuration, HTTP execution via `AgentRunner`, durability behavior, and inspection with Diagrid Dashboard.
* **Introduction docs**: Updated and clarified the benefits and positioning of Dapr Agents, including durable execution, agent identity, and workflow-backed behavior.
* **Quickstarts docs**: Updated the introduction to reference the new Dapr Agents Fundamentals quickstart instead of the older Hello World example.

## Issue reference

<!--Please reference the issue this PR will close: #[issue number]-->

bibryam committed:
Updated Dapr Agents docs to align with recent quickstart changes

Signed-off-by: Bilgin Ibryam <bibryam@gmail.com>

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
+        # Configure the agent to use Dapr State Store for conversation history.
+        memory=AgentMemoryConfig(
+            store=ConversationDaprStateMemory(
+                store_name="conversation-statestore",
```
All the names here should match those that we agree as the default names. Should this not be "agent-memory" ?

bibryam replied:
This doc is aligned with the quickstart. 
Default component names we introduce recently and quickstarts are not updated with them.
When the quickstarts are updated with the default component names, we'll update the doc as well.

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
+* `06_durable_agent_http.py` – the main durable weather agent application exposed over HTTP
+* `function_tools.py` – contains `slow_weather_func`, the tool used by the agent
+* `resources/llm-provider.yaml` – Conversation API and LLM configuration
+* `resources/conversation-statestore.yaml` – conversation memory state store
```
Should this be called conversation or just memory store? And where does the term session come into this?

bibryam replied:
Sessions are like the different threads in ChatGPT left panel.All sessions are stored in the conversation memory.

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
+            )
+        ),
+        # This is where the execution state is stored
+        state=AgentStateConfig(
```
I actually think this class should be renamed to AgentWorkflowConfig() and not AgentStateConfig() so it is explicit

bibryam replied:
Note taken. But this will be done separately from this docs PR.

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
````diff
@@ -51,162 +51,326 @@ docker ps
+),
+```
+
+This tells the agent to store conversation history in a Dapr state store named `conversation-statestore`, under a given `session_id`. The matching Dapr component is `resources/conversation-statestore.yaml`:
````
See comment above about renaming this to memory-statestore

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
 kind: Component
 metadata:
-  name: openai
+  name: conversation-statestore
```
We should use the same as the default names here IMO. So `agent-memory`

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: workflow-statestore
```
We should use the same as the default names here IMO. So `agent-workflow` (although I do prefer `agent-workflow-statestore` to be more explicit.

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
-def get_weather() -> str:
-    """Get current weather."""
-    return "It's 72°F and sunny"
+* This is again a Redis-backed store holds durable workflow state.
```
```suggestion
* This is another Redis state store that holds the durable workflow state.
```

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-introduction.md:
```diff
@@ -11,12 +11,13 @@ aliases:
 
 
-Dapr Agents is a developer framework for building durable and resilient AI agent systems powered by Large Language Models (LLMs). Built on the battle-tested Dapr project, it enables developers to create autonomous systems that reason through problems, make dynamic decisions, and collaborate seamlessly. It includes built-in observability and stateful workflow execution to ensure agentic workflows  ...
+Dapr Agents is a developer framework for building durable and resilient AI agent systems powered by Large Language Models (LLMs). Built on the battle-tested Dapr project, it enables developers to create autonomous systems that have identity, reason through problems, make dynamic decisions, and collaborate seamlessly. It includes built-in observability and stateful workflow execution to ensure age ...
```
Looks to be a grammar mistake here
"for intellI want to organise only the section below. igent, adaptive "

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-introduction.md:
```diff
@@ -25,28 +26,27 @@ Dapr Agents is a developer framework for building durable and resilient AI agent
+| [**MCP Support**]({{% ref "dapr-agents-core-concepts.md#mcp-support" %}})                             | Built-in support for [Model Context Protocol](https://modelcontextprotocol.io/) enabling agents to dynamically discover and invoke external tools through standardized interfaces.
+| [**Memory Management**]({{% ref "dapr-agents-core-concepts.md#memory" %}})                            | Retain context across interactions with options from simple in-memory lists to vector databases, integrating with [Dapr state stores]({{% ref state-management-overview.md %}}) for scalable, persistent memory.
+| [**Durable Agents**]({{% ref "dapr-agents-core-concepts.md#durable-agents" %}})                       | Workflow-backed agents that provide fault-tolerant execution with persistent state management and automatic retry mechanisms for long-running processes.
+| [**Agent Runner**]({{% ref "dapr-agents-core-concepts.md#agent-runner" %}})                           | Expose agents over HTTP or subscribe to a PubSub for long-running tasks, enabling API access to agents without requiring a user interface or human intervention.
```
I could not find a good example of what the Agent Runner does? Must it be 1:1 to an agent, or can it have many agents associated with it? I think you have to have much more than this in the docs.

bibryam replied:
The details on Agent Runner will be in the Core Concept page. Here is just a link to Agent Runner.

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
+        ),
+    )
+
+    # This runner will run the agent and expose it on port 8001
```
I would love to have a longer more detailed description of what a runner can do here. in particular is this 1:1 to an agent?

msfussell commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
```diff
@@ -51,162 +51,326 @@ docker ps
+        ),
+    )
+
+    # This runner will run the agent and expose it on port 8001
```
```suggestion
    # The AgentRunner expose the weather agent on HTTP port 8001 using the serve capability. You could also expose the weather agent as standalone or to send message with XXX" <- Complete this
```

bibryam replied:
Done

    # AgentRunner exposes the weather agent over HTTP on port 8001 using serve.
    # The same runner supports PubSub subscriptions and direct in-process invocation.

msfussell reviewed (commented):
This needs another pass to make it consistent with component naming defaults

msfussell committed:
Merge branch 'v1.16' into dapr-agents-quickstarts

bibryam committed:
Update daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Bilgin Ibryam <bibryam@gmail.com>

bibryam committed:
Address feedback from PR

Signed-off-by: Bilgin Ibryam <bibryam@gmail.com>

bibryam commented:
Aligned the doc with the current state of the quick start and addressed the comments.
More changes will happen to these docs once the quick starts are updated. 
@msfussell 
At this point, I think this PR is good to be merged based on the current state of the quick starts.

msfussell committed:
Merge branch 'v1.16' into dapr-agents-quickstarts

msfussell reviewed (approved):
LGTM

marcduiker commented on daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md:
````diff
@@ -51,162 +51,327 @@ docker ps
 ```
 
-This command starts a Dapr sidecar with the conversation component and launches the agent that communicates with the sidecar for state persistence. Notice how in the agent's responses, it remembers the user's name from the first chat interaction, demonstrating the conversation memory in action.
+Open the dashboard in a browser at [http://localhost:8080](http://localhost:8080) to explore your local workflow executions.
````
```suggestion
Open the dashboard in a browser at `http://localhost:8080` to explore your local workflow executions.
```

bibryam committed:
Update daprdocs/content/en/developing-ai/dapr-agents/dapr-agents-getting-started.md

Co-authored-by: Marc Duiker <marcduiker@users.noreply.github.com>
Signed-off-by: Bilgin Ibryam <bibryam@gmail.com>
